content/en/admin/migrate-to-s3.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Migrate assets to S3
+description: Instructional guide on migrating assets and images to Amazon S3 during production.
+menu:
+  docs:
+    weight: 130
+    parent: admin
+---
+
+## Migrating locally-hosted files to Amazon S3 {#migrate-local-to-s3}
+
+The most expensive part of hosting Mastodon is storing assets and images on the same production server the main Mastodon process is running on. You may have set up the Mastodon server with the assets being hosted locally, but you now want to migrate to [Amazon S3](https://aws.amazon.com/) for flexible storage pricing.
+
+### Pre-requisites {#pre-requisites}
+
+You will need:
+
+#### s3cmd {#s3cmd}
+
+```bash
+sudo apt-get install s3cmd
+```
+
+#### AWS Access keys {#access-keys}
+
+Check your admin console for these.
+
+### Configuring s3cmd {#configure-s3cmd}
+
+To configure s3cmd, run the following as the `mastodon` user:
+
+```bash
+s3cmd --configure
+```
+
+You can test if the configuration works with the following (replace `$S3_BUCKET` with your bucket name):
+
+```bash
+echo "test" > tmp.txt
+s3cmd --acl-public put tmp.txt s3://$S3_BUCKET/tmp.txt
+```
+
+### Uploading the media files {#upload-to-s3}
+
+Stop all of the Mastodon process, switch to the `mastodon` user, and run the following (replacing `$S3_BUCKET` with your bucket name):
```

Comment:
For any setup that isn't brand-new, transferring files to S3 will take a long time. That is a lot of downtime. It's probably better to change config to use S3, restart Mastodon (so new uploads go to S3), then transfer old files. After file transfer is complete, you could use `tootctl media remove-orphans` to clean up transferred files that would've been deleted during this transitionary period.

Reply:
Should be fixed now.